### PR TITLE
Add events to the game.

### PR DESCRIPTION
**Pull request by colin353**

### CODE REVIEW

File: server/state.go [hunk 1]
```diff
@@ -159,12 +159,19 @@ func (s *SiteSelectionController) RecieveMessage(u User, m Message) {
...
 
 func NewSiteVisitController(game *Game) *SiteVisitController {
 	return &SiteVisitController{
-		game: game,
-		name: WaitingState,
+		game:            game,
+		name:            WaitingState,
+		userEventQueue:  map[User][]SiteEvent{},
+		nextMessageID:   0,
+		messageHandlers: map[uint64]SiteEvent{},
```
Comment (iteloo) [inline server/state.go:174]: Should create MessageID alias

File: server/state.go [hunk 2]
```diff
@@ -176,9 +183,39 @@ func (s *SiteVisitController) Begin() {
...
+
+	// Try to give all the users their initial events.
+	for user, _ := range s.game.UserSites {
+		s.GiveNewEvent(user)
+	}
+}
+
+func (s *SiteVisitController) GiveNewEvent(u User) {
+	if len(s.userEventQueue[u]) == 0 {
+		fmt.Printf("No events for %q.", u.Name())
```
Comment (iteloo) [inline server/state.go:202]: Should send placeholder events for now. 
In the future, send randomly-generated effectless events in the future.

Comment (colin353) [reply server/state.go:202]: Cloud

File: server/site_events.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	title := fmt.Sprintf("Repair %s?", site)
+	description := fmt.Sprintf("The %s looks about %d%% functional.", site, g.SiteRepairState[site])
+
+	msg := NewEventMessage(title, description)
+	msg.WithSpendButton(Log)
+	return msg
+}
+func (e RepairSite) End(g *Game, u User, r EventResponseMessage) {
+	site := g.UserSites[u]
+	g.SiteRepairState[site] += uint64(r.ResourceAmount)
```
Comment (iteloo) [inline server/site_events.go:31]: Is this just temporary math?

**Review by iteloo — APPROVED** [review]
Cloud it is

File: server/site_events.go (hunk 3 above)
Comment (colin353) [reply server/site_events.go:31]: I guess there should be a multiplier, at some point.

Comment (iteloo) [reply server/site_events.go:31]: I think the question will come up again when you try to make the status update event

**Comment by iteloo** [comment]
lgtm

**Pull request merged**